File: backend/api/src/main/java/kpn/api/common/RouteLocationAnalysis.java
package kpn.api.common;

import kpn.api.common.location.Location;
import kpn.api.common.location.LocationCandidate;

import java.util.Optional;
import com.google.common.collect.ImmutableList;

public record RouteLocationAnalysis(
  Optional<Location> location,
  ImmutableList<LocationCandidate> candidates,
  ImmutableList<String> locationNames
) {
}

File: backend/api/src/main/scala/kpn/api/common/RouteLocationAnalysis.scala
package kpn.api.common

import kpn.api.common.location.Location
import kpn.api.common.location.LocationCandidate

case class RouteLocationAnalysis(
  location: Option[Location],
  candidates: Seq[LocationCandidate],
  locationNames: Seq[String]
)
